File: Fourth Semester/Systems-for-Design-and-Implementation/Lab04/Lab04-C#/Lab04/domain/ProbeParticipant.cs
namespace Lab03.domain;

public class ProbeParticipant : Identifiable<int>
{
    private Participant participant;

    private Proba proba;

    private int id;

    public ProbeParticipant(Participant participant, Proba proba, int id)
    {
        this.participant = participant;
        this.proba = proba;
        this.id = id;
    }


    public void setId(int id)
    {
        throw new NotImplementedException();
    }

    public int getId()
    {
        throw new NotImplementedException();
    }

    public Participant Participant
    {
        get => participant;
        set => participant = value ?? throw new ArgumentNullException(nameof(value));
    }

    public Proba Proba
    {
        get => proba;
        set => proba = value ?? throw new ArgumentNullException(nameof(value));
    }

    public int Id
    {
        get => id;
        set => id = value;
    }
}

File: Fourth Semester/Systems-for-Design-and-Implementation/Lab05-Java/demo/src/main/java/com/example/demo/domain/ProbeParticipant.java
package com.example.demo.domain;

public class ProbeParticipant implements Identifiable<Integer>{

    private Integer id;
    private Participant participant;
    private Proba proba;

    public ProbeParticipant(Integer id, Participant participant, Proba proba) {
        this.id = id;
        this.participant = participant;
        this.proba = proba;
    }


    @Override
    public void setId(Integer integer) {

    }

    @Override
    public Integer getId() {
        return null;
    }

    public Participant getParticipant() {
        return participant;
    }

    public Proba getProba() {
        return proba;
    }

    public void setParticipant(Participant participant) {
        this.participant = participant;
    }

    public void setProba(Proba proba) {
        this.proba = proba;
    }
}
